### PR TITLE
Use @Upsert from Room instead of a custom helper

### LINKED ISSUE #243: [FR] Use @Upsert from Room
Version [`2.5.0-alpha03`](https://developer.android.com/jetpack/androidx/releases/room#2.5.0-alpha03) of Room added an `@Upsert` annotation, which we probably should be able to use instead of our own custom upsert helper.

### PR DESCRIPTION
**Pull request by yveskalume**
Fix #243

### CODE REVIEW

File: core/database/src/main/java/com/google/samples/apps/nowinandroid/core/database/dao/AuthorDao.kt [hunk 1]
```diff
@@ -56,12 +56,8 @@ interface AuthorDao {
     /**
      * Inserts or updates [entities] in the db under the specified primary keys
      */
-    @Transaction
-    suspend fun upsertAuthors(entities: List<AuthorEntity>) = upsert(
```
Comment (alexvanyo) [inline core/database/src/main/java/com/google/samples/apps/nowinandroid/core/database/dao/AuthorDao.kt:60]: I believe we should be able to remove the now-unused `upsert` method as part of this PR.

**Review by alexvanyo — COMMENTED** [review]
Thanks for contributing! One request to remove the code that will be unused as part of this, and also to fix up the tests that are failing.

**Review by alexvanyo — COMMENTED** [review]
Thanks for contributing! One request to remove the code that will be unused as part of this, and also to fix up the tests that are failing.

**Comment by yveskalume** [comment]
> Thanks for contributing! One request to remove the code that will be unused as part of this, and also to fix up the tests that are failing.

thanks for the review, what should i put in the tests ?

**Review by tunjid — APPROVED** [review]
Thanks for doing this! It looks great! Just waiting on CI to pass.

**Pull request merged**